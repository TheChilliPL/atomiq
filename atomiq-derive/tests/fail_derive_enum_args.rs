#![cfg(not(clippy))]

use atomiq_derive::Atomizable;

#[derive(Atomizable)]
#[repr(u8)]
enum TestEnum {
    A,
    B(u8),
    C { value: u8 },
}

fn main() {}