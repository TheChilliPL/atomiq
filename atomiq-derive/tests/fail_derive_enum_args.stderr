error: Atomizable can only be derived for enums with only unit variants.
 --> tests/fail_derive_enum_args.rs:7:6
  |
7 |     B(u8),
  |      ^^^^

error: Atomizable can only be derived for enums with only unit variants.
 --> tests/fail_derive_enum_args.rs:8:7
  |
8 |     C { value: u8 },
  |       ^^^^^^^^^^^^^
